### DISCUSSION TITLE
ValueError upon model initialization

### DISCUSSION THREAD

**Discussion by np-n** [mindee/doctr#916] [Q&A] [answered]
I am always getting the exception `ValueError: Tensor("Const_1:0", shape=(3,), dtype=float32) must be from the same graph as Tensor("stack:0", shape=(1, 1024, 1024, 3), dtype=float32) (graphs are <tensorflow.python.framework.ops.Graph object at 0x7f7a8cd8ee50> and <tensorflow.python.framework.ops.Graph object at 0x7f7a180613d0>).
`  while initializing the model using `model = ocr_predictor(pretrained=True)`. 

How to handle the above exception? When I started to debug, I found that it is unable to get repr. i.e. `Unable to get repr for <class 'doctr.models.predictor.tensorflow.OCRPredictor>`. I have described more at :https://stackoverflow.com/questions/72210187/unable-to-get-repr-for-class-doctr-models-predictor-tensorflow-ocrpredictor

**Comment by felixdittrich92** [comment]
Hi @np-n 👋 ,
Can you please post your code and the output from: https://github.com/mindee/doctr/blob/main/scripts/collect_env.py ? :)
Edit: and the shape from your algned_image please

**Reply by np-n** [reply]
The aligned image is the 3-dimensional array. Its shape is (618,919,3).

**Reply by felixdittrich92** [reply]
@np-n 
I'm not able to reproduce your behaviour:

```
import numpy as np
from doctr.models import ocr_predictor

dummy = np.random.rand(618,919,3).astype(np.float32)
model = ocr_predictor(pretrained=True)
x = model([dummy])
print(x)
```

please post the output from `collect_env.py` script and try to reinstall with `pip install "python-doctr[tf]" --upgrade` :)
Edit: tested with 0.5.1 and dev (0.6.0)

**Reply by np-n** [reply]
This is output from `collect_env.py` script.
![output](https://user-images.githubusercontent.com/39429615/168011217-ba93b9a1-6c0c-4f8f-a6f2-8ce3a48ecb6c.png)

**Reply by felixdittrich92** [reply]
mh looks good have you tried to reinstall ?

**Reply by np-n** [reply]
yes, I have done everything!

**Reply by felixdittrich92** [reply]
please try:
```
import tensorflow as tf
tf.keras.backend.clear_session()
model = ocr_predictor(pretrained=True)
x = model([aligned_image])
```
This is really strange :sweat_smile:

**Reply by np-n** [reply]
Yes, it works. Could you help me by explaining how it works after clearing the backend session? Thank you

**Reply by felixdittrich92** [reply]
`(graphs are <tensorflow.python.framework.ops.Graph object at 0x7f7a8cd8ee50> and <tensorflow.python.framework.ops.Graph object at 0x7f7a180613d0>)` indicates, that there is already a Graph object into memory after clearing the session this cannot be the problem :)
I cannot explain why this happens (has nothing to do with docTR maybe a good issue for the TF repo) but i have noticed this quite often, especially when writing test cases for docTR :sweat_smile:

**Reply by np-n** [reply]
Thank you so much for your help!. Now, I am wondering to know, how to train DoCTR for the different languages. I am looking for an OCR for Devanagari scripts. Could you please guide me through the approach that I should follow in order to train it for the Devanagari Language?

**Comment by frgfm** [comment, marked as answer]
For people looking for solutions, as suggested by Felix [here](https://github.com/mindee/doctr/discussions/916#discussioncomment-2737326), the problem was coming from keras graph referencing in a loop. Resetting the session index of keras should do the trick :)
